modules/aws/agentless-gw/main.tf

```diff
@@ -42,19 +42,27 @@ module "gw_instance" {
   ami                               = var.ami
   instance_profile_name             = var.instance_profile_name
   additional_install_parameters     = var.additional_install_parameters
-  password                          = var.password
-  password_secret_name              = var.password_secret_name
+  admin_password                    = var.admin_password
+  secadmin_password                 = var.secadmin_password
+  sonarg_password                   = var.sonarg_password
+  sonargd_password                  = var.sonargd_password
+  admin_password_secret_name        = var.admin_password_secret_name
+  secadmin_password_secret_name     = var.secadmin_password_secret_name
+  sonarg_password_secret_name       = var.sonarg_password_secret_name
+  sonargd_password_secret_name      = var.sonargd_password_secret_name
   ssh_key_path                      = var.ssh_key_pair.ssh_private_key_file_path
   binaries_location                 = var.binaries_location
+  tarball_url                       = var.tarball_url
   hub_sonarw_public_key             = var.hub_sonarw_public_key
   hadr_dr_node                      = var.hadr_dr_node
   main_node_sonarw_public_key       = var.main_node_sonarw_public_key
   main_node_sonarw_private_key      = var.main_node_sonarw_private_key
-  proxy_info                        = var.ingress_communication_via_proxy
+  proxy_info                        = var.gw_proxy_info
   skip_instance_health_verification = var.skip_instance_health_verification
   terraform_script_path_folder      = var.terraform_script_path_folder
-  use_public_ip                     = false
+  use_public_ip                     = var.use_public_ip
```

Comment:
If you expose use_public_ip, don't you need to expose also attach_persistent_public_ip ?

Reply:
But if you already decide to provide an option to set a public IP to the GW, maybe at other use cases someone else would like to have the option to set also a persistent public IP

Reply:
I'm still not quite sure what you mean, this is just related to what IP to use when running ssh.